engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/FlinkMiniClusterTableOperations.scala

```diff
@@ -179,3 +183,10 @@ object FlinkMiniClusterTableOperations extends LazyLogging {
   }
 
 }
+
+sealed trait DataGenerationMode
+
+object DataGenerationMode {
+  case object RandomDataMode                                                    extends DataGenerationMode
+  case class RealDataMode(sqlStatements: List[SqlStatement], tableName: String) extends DataGenerationMode
```

Comment:
Modes are rather things that are parameterless enums. Let's reserve this word for this purpose and use something else here. Also, I see that you use it very locally now, you could just have two methods in `FlinkMiniClusterTableOperations`

Reply:
It looks much more intuitive now :+1: